src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -718,7 +718,7 @@ public class StripePaymentService : IPaymentService
                 ProrationDate = prorationDate,
             };
 
-            if ((subscriptionUpdate.SubscriptionItem(sub)?.Quantity ?? 0) == updatedItemOptions.Quantity)
+            if (subscriptionUpdate.UpdateNeeded(sub))
```

Comment:
This case seems wrong? The comment below mentions no need to update, but we the code says an update is needed.